index.js

```diff
@@ -0,0 +1,129 @@
+const express = require('express');
+const bcrypt = require('bcryptjs');
+const session = require('express-session');
+const cors = require('cors');
+const KnexSessionStore = require('connect-session-knex')(session);
+const db = require('./data/helpers');
+const knexDb = require('./data/dbConfig');
+
+const server = express();
+
+const sessionConfig = {
+    name: 'secretmysterycookie',
+    secret: 'lasdn3039fff$#$RGfgd',
+    cookie: {
+        maxAge: 1000 * 60 * 10,
+        secure: false
+    },
+    httpOnly: true,
+    resave: false,
+    saveUninitialized: false,
+    store: new KnexSessionStore({
+        tablename: 'sessions',
+        sidfieldname: 'sid',
+        knex: knexDb,
+        createtable: true,
+        clearInterval: 1000 * 60 * 60
+    })
+};
+
+// Middleware
+server.use(session(sessionConfig));
+server.use(express.json());
+server.use(cors());
+
+// Custom Middleware
+function protected(req, res, next) {
+    if (req.session && req.session.userId) {
+        next();
+    } else {
+        res.status(401).json({ message: 'You are not logged in.' });
+    }
+}
+
+// REGISTER EndP01ntZ
+server.post('/api/register', async(req, res) => {
+    // get form input values
+    const registrationData = req.body;
+
+    // validate data
+    if (!registrationData.username || !registrationData.password) {
+        return res
+            .status(400)
+            .json({ message: 'Please enter a valid username and password.' });
+    }
+
+    // Check if user already exists
+    try {
+        const userInDb = await db.getByUsername(registrationData.username);
+        if (userInDb) {
+            res.status(422).json({ message: 'That username already exists.' });
+        }
+    } catch (error) {
+        res.status(500).json({ error: 'caught' });
+    }
+
+    // generate a hash
+    const hash = bcrypt.hashSync(registrationData.password, 10);
+
+    // replace plain text with hash
+    registrationData.password = hash;
+
+    // save to db
+    try {
+        const newUserId = await db.addUser(registrationData);
```

Comment:
You are trying to call a function `addUser` which does not exist, because of that it is impossible to add new user.